### PR TITLE
feat: add report page

### PR DESCRIPTION
**Pull request by LizavetaY**
### Description

Please provide a brief description of the changes you have made in your PR.

### Type of change

- [x] Feature (introduces new functionality)
- [ ] Bugfix
- [x] Configuration
- [ ] Refactoring (doesn't change functionality, only the code)
- [ ] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)
- [x] Chakra components/constants/hooks have been used wherever possible

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

Add screenshots from the [current deployed state](https://sandbox-money-app.herokuapp.com/) of the changed components here.

#### Screenshots/Video after change

https://user-images.githubusercontent.com/47990337/185791620-f64c0d1f-5c95-4402-9616-12443aa15d04.mov

https://user-images.githubusercontent.com/47990337/185791632-ce4bed19-425e-4820-990d-2cf1fc7bcc5c.mov

https://user-images.githubusercontent.com/47990337/185791623-9b22bf30-9c57-4de8-bd98-c1ea7218494c.mov

https://user-images.githubusercontent.com/47990337/185791625-180c5701-f85e-4736-9388-36710d27b9d3.mov

https://user-images.githubusercontent.com/47990337/185791626-c9527b22-225a-422f-98ac-da495e8bcec0.mov

### CODE REVIEW

File: src/components/FiltersReport/utils.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    case 'thisMonth':
+      return i18next.t('report.filters.date.value.thisMonth');
+
+    case 'lastMonth':
+      return i18next.t('report.filters.date.value.lastMonth');
+
+    case 'customized':
+      return i18next.t('report.filters.date.value.customized');
+
```
Comment (itsnotpolza) [inline src/components/FiltersReport/utils.js:117]:
```suggestion
export const getInputValueName = (value) => {
  return i18next.t(`report.filters.date.value.${value}`)
 }
```

Comment (itsnotpolza) [reply src/components/FiltersReport/utils.js:117]: not even sure you need this function to be honest
]

File: src/helpers/filtersHelpers.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    switch (true) {
+      case isCategoryIncomeType:
+        return 'categoryIncomeReport';
+
+      case isCategoryExpenseType:
+        return 'categoryExpenseReport';
+
+      case isPayerType:
+        return 'payerReport';
+    }
```
Comment (itsnotpolza) [inline src/helpers/filtersHelpers.js:35]:
```suggestion
     if(!!getCategoriesOptions(
      categoriesData,
      'Income'
    )?.find((category) => category.value === data[0]?.value)) {
        return 'categoryIncomeReport';
    };
    if(!!getCategoriesOptions(
      categoriesData,
      'Expense'
    )?.find((category) => category.value === data[0]?.value)) {
       return 'categoryExpenseReport';
    }
    if(!!getPayersOptions(payersData)?.find(
      (payer) => payer.value === data[0]?.value
    )){ 
       return 'payerReport';
    };
```

File: src/components/FiltersReport/utils.js (hunk 1 above)
Comment (LizavetaY) [reply src/components/FiltersReport/utils.js:117]: Снимок экрана 2022-08-21 в 8 16 25 PM

File: src/helpers/filtersHelpers.js (hunk 2 above)
Comment (LizavetaY) [reply src/helpers/filtersHelpers.js:35]: Снимок экрана 2022-08-21 в 8 42 13 PM

**Review by itsnotpolza — APPROVED** [review]

**Review by HarryPotter-4000 — APPROVED** [review]

**Pull request merged**